### DISCUSSION TITLE
Async SQLAlchemy and load_only

### DISCUSSION THREAD

**Discussion by Akashkarmokar** [sqlalchemy/sqlalchemy#11930] [Usage Questions] [answered]
Here are my  Model declaration with simple attribute : 

```
class Register(TimestampedModel):
    __tablename__ = 'registers'

    id: Mapped[int] = mapped_column(primary_key=True)
    email: Mapped[str] = mapped_column(nullable=False)
    password: Mapped[str] = mapped_column(nullable=False)
    profile = relationship("Profile",back_populates='register_user')
```
    
    
```
   
class Profile(TimestampedModel):
    __tablename__ = 'profiles'

    id: Mapped[int] = mapped_column(primary_key=True)
    name: Mapped[Text] = mapped_column(Text,nullable=True)

    # Foreign Keys
    register_user_id: Mapped[int] = mapped_column(ForeignKey('registers.id'))


    # Relationships 
    register_user: Mapped["Register"] = relationship("Register",back_populates='profile')
```

and my route to fetch data from DB : 

```
@AuthRoutes.post("/practice")
async def practice(session: AsyncSession):

    try:
        stmt = select(Register).options(
                selectinload(Register.profile),
                load_only(Register.id, Register.email)
            )
      
        async with session.begin() as session: 
        
            result = await session.execute(stmt)
            result = result.unique()
            users = result.scalars().all()


            user_schema = RegisterSchema(many= True)
            users_dict = user_schema.dump(users)
            

        return {
            'message': 'ok',
            'data': users_dict
        }

    except Exception as err:
        print("Error: ", err)
        return {
            'message': 'bad',
            'data': []
        }
```
        

So the problem I am getting is that if I use load_only function to reduce column it's not working and send me this error: "Error:  greenlet_spawn has not been called; can't call await_only() here. Was IO attempted in an unexpected place? (Background on this error at: https://sqlalche.me/e/20/xd2s)". 
If I omit load_only it works perfectly.

**Comment by CaselIT** [comment, marked as answer]
Hi,

your `user_schema.dump(users)` is trying to access the missing columns, that are not loaded. On access sqlalchemy will try to load from the db, but with asyncio you can't do that (also you really wouldn't what that event in sync code).

Answer here is change your serializer function to only take into consideration the loaded attributes. maybe you can pass the attributes to serialize in the constructor `RegisterSchema(many= True)`? 

Also `result = result.unique()` is not needed if you don't use joinedload
